library/spec/Ivoz/Core/Infrastructure/Domain/Service/Lifecycle/DoctrineEventSubscriberSpec.php

```diff
@@ -0,0 +1,153 @@
+            $this->eventPublisher,
+            $this->commandPersister,
+            $this->forcedEntityChangeLog
+        );
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(DoctrineEventSubscriber::class);
+    }
+
+    function it_triggers_entity_specific_error_handler()
+    {
+        $errorEvent = $this->getErrorEvent();
+
+        $handlerServiceName =  'provider.lifecycle.company.error_handler';
+        $handlerService = $this->getTestDouble(
+            \Ivoz\Core\Domain\Service\PersistErrorHandlerInterface::class
+        );
+        /** @var PersistErrorHandlerServiceCollection $handlerCollection */
+        $handlerCollection = $this->getInstance(
+            PersistErrorHandlerServiceCollection::class
+        );
+        $handlerCollection->setServices([$handlerService->reveal()]);
+
+        $this->getterProphecy(
+            $this->serviceContainer,
+            [
+                'has' => function () use ($handlerServiceName) {
+                    return [[$handlerServiceName], true];
+                },
+                'get' => function () use ($handlerServiceName, $handlerCollection) {
+                    return [[$handlerServiceName], $handlerCollection];
+                }
+            ]
+        );
+
+        $this
+            ->serviceContainer
+            ->get('lifecycle.common.error_handler')
+            ->shouldNotBeCalled();
+
+        $handlerService
+            ->handle(Argument::type(\Exception::class))
+            ->shouldBeCalled()
+            ->willThrow(new \DomainException('expected response'));
+
+        $this
+            ->shouldThrow('\DomainException')
+            ->during('onError', [$errorEvent]);
+    }
+
+    function it_triggers_common_error_handler_as_fallback()
+    {
+        $errorEvent = $this->getErrorEvent();
+
+        $handlerServiceName =  'provider.lifecycle.company.error_handler';
+        /** @var PersistErrorHandlerServiceCollection $handlerCollection */
+        $handlerCollection = $this->getInstance(
+            PersistErrorHandlerServiceCollection::class
+        );
+
+        $this->getterProphecy(
+            $this->serviceContainer,
+            [
+                'has' => function () use ($handlerServiceName) {
+                    return [[$handlerServiceName], true];
+                },
+                'get' => function () use ($handlerServiceName, $handlerCollection) {
+                    return [[$handlerServiceName], $handlerCollection];
+                }
+            ]
+        );
+
+        $commonHandlerCollection = $this->getTestDouble(
+            PersistErrorHandlerServiceCollection::class,
+            false
+        );
+
+        $commonHandlerCollection
+            ->execute(Argument::any())
+            ->shouldBeCalled()
+            ->willThrow(new \RuntimeException('runtime exception'));
+
+        $this
+            ->serviceContainer
+            ->get('lifecycle.common.error_handler')
```

Comment:
Can this be replaced by CommonLifecycleServiceCollection::class after #974 has been merged?

Reply:
It will break otherwise :)

Reply:
pity, i like constants :(